src/Bedrock.Framework.Experimental/Protocols/Kafka/SequenceReaderExtensions.cs

```diff
@@ -0,0 +1,138 @@
+﻿#nullable enable
+
+using Bedrock.Framework.Experimental.Protocols.Kafka.Models;
+using System;
+using System.Buffers;
+using System.Text;
+
+namespace Bedrock.Framework.Experimental.Protocols.Kafka
+{
+    public static class SequenceReaderExtensions
+    {
+        public static string? ReadNullableString(this ref SequenceReader<byte> reader)
+        {
+            if (!reader.TryReadBigEndian(out short length))
```

Comment:
Should all of these be `TryRead___` ?? (solves the TODOs)
Is your theory that `KafkaMessageReader` has ensured that the whole message is read, so you don't need to "Try" at this point?